### PR TITLE
[Utopia 769] Updated pia-intake entity and associated JSONB classes and dtos

### PR DESCRIPTION
**Pull request by kushal-arora-fw**
# Description

This PR includes the following proposed change(s):

Updated pia-intake entity and associated jsonb classes and dtos for
- collectionUseAndDisclosure
- StoringPersonalInformation

CollectionUseAndDisclosure empty initial default state. For filled in information, refer Tests.
```
"collectionUseAndDisclosure": {
    "steps": [],
    "collectionNotice": {
      "drafterInput": null,
      "mpoInput": null
    }
  },
```

StoringPersonalInformation Empty initial default state. For filled in information, refer Tests.
```
"storingPersonalInformation": {
    "personalInformation": {
      "storedOutsideCanada": null,
      "whereDetails": null
    },
    "sensitivePersonalInformation": {
      "doesInvolve": null,
      "disclosedOutsideCanada": null
    },
    "disclosuresOutsideCanada": {
      "storage": {
        "sensitiveInfoStoredByServiceProvider": null,
        "serviceProviderList": [],
        "disclosureDetails": null,
        "contractualTerms": null,
      },
      "contract": {
        "relyOnExistingContract": null,
        "enterpriseServiceAccessDetails": null,
      },
      "controls": {
        "unauthorizedAccessMeasures": null,
      },
      "trackAccess": {
        "trackAccessDetails": null,
      },
      "risks": {
        "privacyRisks": [],
      },
    },
  }

```

securityPersonalInformation: Empty initial default state. For filled in information, refer Tests.
```
"securityPersonalInformation": {
    "digitalToolsAndSystems": {
      "toolsAndAssessment": {
        "involveDigitalToolsAndSystems": null,
        "haveSecurityAssessment": null
      },
      "storage": {
        "onGovServers": null,
        "whereDetails": null
      }
    },
    "accessToPersonalInformation": {
      "onlyCertainRolesAccessInformation": null,
      "accessApproved": null,
      "useAuditLogs": null,
      "additionalStrategies": null
    }
  }
```

accuracyCorrectionAndRetention: Empty initial default state. For filled in information, refer Tests.
```
"accuracyCorrectionAndRetention": {
    "accuracy": {
      "description": null
    },
    "correction": {
      "haveProcessInPlace": null,
      "willDocument": null,
      "willConductNotifications": null
    },
    "retention": {
      "usePIForDecision": null,
      "haveApprovedInfoSchedule": null,
      "describeRetention": null
    }
  }
```

personalInformationBanks: Empty initial default state. For filled in information, refer Tests.
```
  "personalInformationBanks": {
    "resultingPIB": {
      "willResultInPIB": null,
      "descriptionInformationType": null,
      "mainMinistryInvolved": null,
      "managingPersonName": null,
      "managingPersonPhone": null
    }
  }
```
additionalRisks: Empty initial default state. For filled in information, refer Tests.
```
"additionalRisks": {
    "risks": []
  }
```
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [x] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [x] New and existing unit tests pass locally with my changes
- [x] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/backend/src/modules/pia-intake/jsonb-classes/collection-use-and-disclosure/steps-walkthrough.class.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  @IsString()
+  OtherInput?: string;
+}
+
+export const stepWalkthroughDetails: Array<IFormField<StepWalkthrough>> = [
+  {
+    key: 'drafterInput',
+    type: 'text',
+    isRichText: false,
+    allowedUserTypesEdit: null, // any
```
Comment (binggaofw) [inline src/backend/src/modules/pia-intake/jsonb-classes/collection-use-and-disclosure/steps-walkthrough.class.ts:24]: not understanding the comments  '// any' what does it mean?

Comment (kushal-arora-fw) [reply src/backend/src/modules/pia-intake/jsonb-classes/collection-use-and-disclosure/steps-walkthrough.class.ts]: the idea is any role in the system can access and edit this field. will place restrictions soon in the json when we will use this field.

Comment (kushal-arora-fw) [reply src/backend/src/modules/pia-intake/jsonb-classes/collection-use-and-disclosure/steps-walkthrough.class.ts]: updated to a better comment.

**Review by antsand — DISMISSED** [review]

File: src/backend/src/modules/pia-intake/pia-intake.service.ts [hunk 2]
```diff
@@ -417,4 +429,32 @@ export class PiaIntakeService {
...
+   *
+   * @description
+   * This method validates role access to the following fields, if needed:
+   * 1. collectionUseAndDisclosure
+   * 2. storingPersonalInformation
+   * 3. securityPersonalInformation
+   * 4. accuracyCorrectionAndRetention
+   * 5. personalInformationBanks
+   * 6. additionalRisks
+   */
```
Comment (binggaofw) [inline src/backend/src/modules/pia-intake/pia-intake.service.ts:449]: so this pr just validate the first tab ?

**Review by binggaofw — DISMISSED** [review]

File: src/backend/src/modules/pia-intake/pia-intake.service.ts (hunk 2 above)
Comment (kushal-arora-fw) [reply src/backend/src/modules/pia-intake/pia-intake.service.ts:449]: clarified over the call. At the moment, we only have role checks in first tab.

**Review by binggaofw — DISMISSED** [review]

**Review by binggaofw — APPROVED** [review]

**Pull request merged**